Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,492 @@
+    def _on_err(self, err_msg):
+        self._logger.error(err_msg)
+
+
+class ChannelSource(BoostingSource):
+    """
+    Credit mining source from a channel.
+    """
+    def __init__(self, session, dispersy_cid, boost_settings, torrent_insert_cb):
+        BoostingSource.__init__(self, session, dispersy_cid, boost_settings, torrent_insert_cb)
+
+        self.channel_id = None
+
+        self.channel = None
+        self.community = None
+        self.database_updated = True
+
+        self.check_torrent_interval = 10
+        self.dispersy_cid = dispersy_cid
+
+        self.session.add_observer(self._on_database_updated, NTFY_TORRENTS, [NTFY_INSERT, NTFY_UPDATE])
+
+        d = self._load_if_ready(dispersy_cid)
+        self.boosting_manager.register_task(str(self.source) + "_load", d, value=dispersy_cid)
+
+        self.unavail_torrent = {}
+
+    def kill_tasks(self):
+        BoostingSource.kill_tasks(self)
+
+        self.session.remove_observer(self._on_database_updated)
+
+    def _load(self, dispersy_cid):
+        dispersy = self.session.get_dispersy_instance()
+
+        @blocking_call_on_reactor_thread
+        def join_community():
+            """
+            find the community/channel id, then join
+            """
+            try:
+                self.community = dispersy.get_community(dispersy_cid, True)
+                self.boosting_manager.register_task(str(self.source) + "_get_id",
+                                                    reactor.callLater(1, get_channel_id))
+
+            except CommunityNotFoundException:
+
+                allchannelcommunity = None
+                for community in dispersy.get_communities():
+                    if isinstance(community, AllChannelCommunity):
+                        allchannelcommunity = community
+                        break
+
+                if allchannelcommunity:
+                    self.community = ChannelCommunity.init_community(dispersy, dispersy.get_member(mid=dispersy_cid),
+                                                                     allchannelcommunity._my_member, self.session)
+                    self._logger.info("Joined channel community %s", dispersy_cid.encode("HEX"))
+                    self.boosting_manager.register_task(str(self.source) + "_get_id",
+                                                        reactor.callLater(1, get_channel_id))
+                else:
+                    self._logger.error("Could not find AllChannelCommunity")
+
+        def get_channel_id():
+            """
+            find channel id by looking at the network
+            """
+            if self.community and self.community._channel_id:
+                self.channel_id = self.community._channel_id
+
+                channel_dict = self.boosting_manager.channelcast_db.getChannel(self.channel_id)
+                self.channel = Channel(*channel_dict)
+
+                if not self.boosting_manager.is_pending_task_active(str(self.source) + "_update"):
```

Comment:
I now see that you are using the TaskManager facilities from the boosting manager instead of making this class a TaskManager subclass. That's gratuitous coupling. Please fix that.